File: examples/csharp/minimal-order-service/Adapters/Outbound/InMemoryOrderRepository.cs
using System.Collections.Concurrent;
using OrderService.Application.Ports.Outbound;
using OrderService.Domain;

namespace OrderService.Adapters.Outbound;

/// <summary>
/// Outbound adapter: In-memory implementation of OrderRepository.
/// Used for testing and simple scenarios.
/// </summary>
public class InMemoryOrderRepository : IOrderRepository
{
    private readonly ConcurrentDictionary<string, Order> _orders = new();

    public void Save(Order order)
    {
        _orders[order.Id] = order;
    }

    public Order? FindById(string id)
    {
        return _orders.TryGetValue(id, out var order) ? order : null;
    }
}

File: examples/java/minimal-order-service/src/main/java/com/example/order/adapters/outbound/InMemoryOrderRepository.java
package com.example.order.adapters.outbound;

import com.example.order.application.ports.outbound.OrderRepository;
import com.example.order.domain.Order;

import java.util.HashMap;
import java.util.Map;

/**
 * Outbound adapter: In-memory implementation of OrderRepository.
 * Used for testing and simple scenarios.
 */
public class InMemoryOrderRepository implements OrderRepository {
    private final Map<String, Order> orders = new HashMap<>();

    @Override
    public void save(Order order) {
        orders.put(order.getId(), order);
    }

    @Override
    public Order findById(String id) {
        return orders.get(id);
    }
}
